jax/_src/pallas/mosaic/lowering.py

```diff
@@ -1793,6 +1793,27 @@ def _neg_lowering_rule(ctx: LoweringRuleContext, x):
 skip_mlir_conversions.add(lax.neg_p)
 
 
+def _sign_lowering_helper(x):
+  if jnp.issubdtype(x.dtype, jnp.unsignedinteger):
+    return (x != 0).astype(x.dtype)
+
+  if jnp.issubdtype(x.dtype, jnp.integer):
+    return (x > 0).astype(x.dtype) - (x < 0).astype(x.dtype)
+
+  if jnp.issubdtype(x.dtype, jnp.floating):
+    out = (x > 0.).astype(x.dtype) - (x < 0.).astype(x.dtype)
+    return jnp.where(jnp.isnan(x), jnp.nan, out)
+
+  raise NotImplementedError
+
+
+def _sign_lowering_rule(ctx: LoweringRuleContext, x):
+  return lower_fun(_sign_lowering_helper, multiple_results=False)(ctx, x)
+
+
+lowering_rules[lax.sign_p] = _sign_lowering_rule
```

Comment:
I think you can do

```
lowering_rules[lax.sign_p] = lower_fun(_sign_lowering_helper, multiple_results=False)
```

and also rename `_sign_lowering_helper` to `_sign_lowering`.

Reply:
I think it's better not to change because all the lowering rules are defined in `lowering_rules[..._p] = _..._lowering_rule` format.

Reply:
Sure, that's okay.

I'm just personally not a big fan of helper functions.